app/services/organizations/importers/csv_import_service.rb

```diff
@@ -4,61 +4,67 @@ module Organizations
   module Importers
     class CsvImportService
       Status = Data.define(:success?, :count, :no_match, :errors)
-      def initialize(file)
+      def initialize(file, current_user_id)
         @file = file
-        @organization = Current.organization
+        @organization = ActsAsTenant.current_tenant
```

Comment:
Will this tennant scope always be reliable with this process running in another thread? I am wondering if we should set the org based off the User.

Reply:
I was concerned of that as well. From the docs:
```
ActsAsTenant supports

    ActiveJob - ActsAsTenant will automatically save the current tenant in ActiveJob arguments and set it when the job runs.

```

I did a "trust but verify" the best I could and it appears it works as it should. I kind of prefer sticking with what is build in with ActsAsTenant than relying on the user. If I was running this service without using a job, I'd want to get the org through ActsAsTenant. 

I added a debugger to the gem to get a better idea how it was working with its active job extension. It basically takes the current tenant and merges it into the job data hash when the job is created and queued. When the job is actually run, it gets the tenant id and sets the current_tenant. The csv service then runs within that scope.

I'll have to eat my words if it breaks in production :grin:

![Screenshot from 2024-12-19 14-24-12](https://github.com/user-attachments/assets/d3a98815-4b72-4c66-860c-cf3b438cbaff)

Reply:
Oh snap, yes you're right - they even have a note in their docs on this. That's really nice. 
https://github.com/ErwinM/acts_as_tenant?tab=readme-ov-file#background-processing-libraries